datafusion/physical-expr/src/planner.rs

```diff
@@ -83,6 +83,84 @@ pub fn create_physical_expr(
                 }
             }
         }
+        Expr::IsTrue(expr) => {
+            let binary_op = binary_expr(
+                expr.as_ref().clone(),
+                Operator::IsNotDistinctFrom,
+                Expr::Literal(ScalarValue::Boolean(Some(true))),
+            );
+            create_physical_expr(
+                &binary_op,
+                input_dfschema,
+                input_schema,
+                execution_props,
+            )
+        }
+        Expr::IsNotTrue(expr) => {
+            let binary_op = binary_expr(
+                expr.as_ref().clone(),
+                Operator::IsDistinctFrom,
+                Expr::Literal(ScalarValue::Boolean(Some(true))),
+            );
+            create_physical_expr(
+                &binary_op,
+                input_dfschema,
+                input_schema,
+                execution_props,
+            )
+        }
+        Expr::IsFalse(expr) => {
+            let binary_op = binary_expr(
+                expr.as_ref().clone(),
+                Operator::IsNotDistinctFrom,
+                Expr::Literal(ScalarValue::Boolean(Some(false))),
+            );
+            create_physical_expr(
+                &binary_op,
+                input_dfschema,
+                input_schema,
+                execution_props,
+            )
+        }
+        Expr::IsNotFalse(expr) => {
+            let binary_op = binary_expr(
+                expr.as_ref().clone(),
+                Operator::IsDistinctFrom,
+                Expr::Literal(ScalarValue::Boolean(Some(false))),
+            );
+            create_physical_expr(
+                &binary_op,
+                input_dfschema,
+                input_schema,
+                execution_props,
+            )
+        }
+        Expr::IsUnknown(expr) => {
```

Comment:
I'm confused about the conversion.
Datafusion support the `IS NULL` or `IS NOT NULL`, why not convert this to `IS_NULL` OR `IS_NOT_NULL`

Reply:
we can convert the `IS UNKNOWN` to `IS NUL` when generating the logical plan.

Reply:
Maybe we can do this in the follow-up pr.

Reply:
I check the spark, and meet a confused result.
1. check with scalar 
```
spark-sql> explain select 1 is unknown;
== Physical Plan ==
org.apache.spark.sql.AnalysisException: cannot resolve '(1 IS UNKNOWN)' due to data type mismatch: argument 1 requires boolean type, however, '1' is of int type.; line 1 pos 17;
'Project [unresolvedalias(anon$1(1), None)]
+- OneRowRelation
```

2. check with column
```
spark-sql> desc t1;
22/08/31 08:22:24 WARN ObjectStore: Failed to get database global_temp, returning NoSuchObjectException
c1                  	int
Time taken: 0.088 seconds, Fetched 1 row(s)
spark-sql> explain select c1 is unknown from t1;
== Physical Plan ==
*(1) Project [isnull(c1#53) AS (c1 IS NULL)#54]
+- Scan hive default.t1 [c1#53], HiveTableRelation [`default`.`t1`, org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe, Data Cols: [c1#53], Partition Cols: []]

```

Reply:
I have checked the Op in the PG: https://www.postgresql.org/docs/current/functions-comparison.html
The data type of `is true, is false..., is unknown...` must be bool type.

Reply:
> we can convert the `IS UNKNOWN` to `IS NUL` when generating the logical plan.

If we do that, then we will lose information. The semantics are different between these two expressions. `IS UNKNOWN` requires a boolean left expression. That is really the intent of the PR; to have a more accurate representation of the query so that we can do better type verification and coercion.

Reply:
@alamb Could you review this one, please? This thread covers the justification for the change.